modules/ServerAPI/src/client/Protocol.coffee

```diff
@@ -775,18 +802,16 @@ class EndpointListController extends AbstractFormController
 		#check whether or not to display time and concentration columns in the endpoint table
 		@showTimeAndConcentration()
 		
-		# Create a list to hold the endpoint controllers in, so we can iterate through them later
-		@endpointControllers = []
-		for lsState in @collection
-			@.addOne(lsState)
-			if @options.readOnly == true
-				#hide remove buttons
-				@$(".bv_remove_row").hide()
 
 		if @options.readOnly == false && @options.newProtocol == false 	#Only render experiments if the protocol is not new and is not read only 
 			@getExperimentsForProtocol()
 		else #if the table isn't rendered, don't render the download files button either
 			@$(".bv_downloadFiles").hide() 
+			if @options.view == "experiment"
+				@getEndpointTable()
+
+			# Need to add handling for rendering endpoint table if it is an experiment editor/browser and not a protocol browser. 
```

Comment:
nit: Looks like you did this so you can remove this commented out bit

Reply:
Good catch, thanks!